internal/xds/translator/route.go

```diff
@@ -108,8 +108,7 @@ func buildXdsRouteMatch(pathMatch *ir.StringMatch, headerMatches []*ir.StringMat
 		} else if pathMatch.SafeRegex != nil {
 			outMatch.PathSpecifier = &routev3.RouteMatch_SafeRegex{
 				SafeRegex: &matcherv3.RegexMatcher{
-					EngineType: &matcherv3.RegexMatcher_GoogleRe2{},
-					Regex:      *pathMatch.SafeRegex,
+					Regex: *pathMatch.SafeRegex,
```

Comment:
shouldn this be `GoogleRe2` instead ?
https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/matcher/v3/regex.proto#envoy-v3-api-msg-type-matcher-v3-regexmatcher

Reply:
According to https://github.com/envoyproxy/envoy/issues/24256#issuecomment-1333007691, 
> google_re2 is required for engine_type while engine_type is optional for RegexMatcher

This PR removes the optional engine_type.

According to https://github.com/envoyproxy/gateway/blob/main/docs/latest/intro/compatibility.rst, we only need to support Envoy 1.25+ so we don't need to worry the old Envoy behavior described in https://github.com/envoyproxy/envoy/issues/24256#issuecomment-1333078240